mlflow/models/utils.py

```diff
@@ -1518,20 +1517,18 @@ def _convert_llm_input_data(data):
     return _convert_llm_ndarray_to_list(data)
 
 
-def _get_temp_file_with_content(file_name: str, content: str, content_format) -> str:
+def _get_temp_file_with_content(temp_dir: str, file_name: str, content: str, content_format) -> str:
```

Comment:
```suggestion
def _get_temp_file_with_content(temp_dir: str, file_name: str, content: Union[str, bytes], content_format) -> str:
```

if `content_format` can be "wb".